apps/sage_intacct/utils.py

```
@@ -1947,3 +1952,44 @@ def get_accounting_fields(self, resource_type: str) -> dict:
 
         return json.loads(json.dumps(response, default=str))
+
+    def search_and_create_vendors(self, workspace_id: int, missing_vendors: list) -> None:
+        """
+        Seach vendors in Intacct and Upsert Vendors in DB
+        :param workspace_id: Workspace ID
+        :param missing_vendors: Missing Vendors List
+        """
+        missing_vendors_batch = [missing_vendors[i:i + 50] for i in range(0, len(missing_vendors), 50)]
+
+        for missing_vendors_batch in missing_vendors_batch:
```

Ruff (0.8.2)

Loop control variable `missing_vendors_batch` overrides iterable it iterates

(B020)